scrapy/utils/misc.py

```diff
@@ -226,7 +226,14 @@ def returns_none(return_node):
         return value is None or isinstance(value, ast.NameConstant) and value.value is None
 
     if inspect.isgeneratorfunction(callable):
-        code = re.sub(r"^[\t ]+", "", inspect.getsource(callable))
+        pattern = r"(^[\t ]+)"
+        src = inspect.getsource(callable)
+        match = re.match(pattern, src)  # Find indentation
+        code = re.sub(pattern, "", src)
```

Comment:
Nitpick 1:
```suggestion
        code = re.sub(pattern, "", src)
        match = re.match(pattern, src)  # find indentation
```

Just swapping the two lines. For me it looks a little bit more readable, in the sense that `match` is only used within the next `if` block. In fact, this would be even better with the walrus operator if we had Python 3.8 as minimum version.

Reply:
Fair point, Thank you! :bow: